src/main.cpp

```diff
@@ -4947,15 +4978,12 @@ std::string GetWarnings(const std::string &strFor)
 //
 
 
-bool AlreadyHave(const CInv &inv) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+bool AlreadyHaveTx(const CInv &inv)
 {
-    AssertLockHeld(cs_main);
```

Comment:
this is definitely a very good thing. reducing the amount of `cs_main` locking is vary important.